app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -123,7 +123,18 @@ public void onNotFound(String key) {
                 });
 
         }
-
+        private Event getNextEvent(List<Event> listEvents) {
+            if (listEvents == null || listEvents.size() == 0) {
+                return null;
+            }
+            Event ret = listEvents.get(0);
```

Comment:
the list of events for the one month is requested from the backend
`.getChapterEventList(homeGdg.getGplusId(),
                    new DateTime(),
                    new DateTime().plusMonths(1))`
. Since I don't know the specifics of the implementations in the backend, I have to go through all the events in the return list to get the closest one in terms of start day and that will be the next event.